src/components/Graph/Graph.tsx

```diff
@@ -220,7 +220,7 @@ const GraphMemo: FC<GraphMemoProps> = memo(
                   key={nodeName}
                   name={nodeName}
                   ref={(node: HTMLDivElement): void => {
-                    nodeRefs.current[nodeName] = node;
+                    if (node) nodeRefs.current[nodeName] = node;
```

Comment:
@radamson's suggestion is valid, although I would suggest we go a step further.

The reason `node` is null is that the callback function passed to `ref` is called every time the ref changes. This happens on mount and unmount. On unmount the element is destroyed so we get `null`. If we only assign the node if present, we end up holding onto the `div` tag that is no longer on the page. This div will now be used in the `getGraphLayout` calculation, even though it’s no longer on the page. This also creates a memory leak, where the browser cannot release the memory used by that div because we are holding onto it.

Instead, if we change it to delete the key out of the `nodeRefs.current` we can continue using `Object.keys(nodeRefs.current)` without the need for any null checks, as every key should have a value.

```suggestion
                    if (node) {
                      nodeRefs.current[nodeName] = node;
                    } else {
                      delete nodeRefs.current[nodeName];
                    }
```

Reply:
That makes sense. Implemented